### PR TITLE
lua: Use @method.call

### PR DESCRIPTION
**Pull request by tomtomjhj**

### CODE REVIEW

File: queries/lua/highlights.scm [hunk 1]
```diff
@@ -172,7 +172,7 @@
 (function_call name: (dot_index_expression field: (identifier) @function.call))
 (function_declaration name: (dot_index_expression field: (identifier) @function))
 
-(method_index_expression method: (identifier) @method)
+(method_index_expression method: (identifier) @method.call)
```
Comment (theHamsta) [inline queries/lua/highlights.scm:175]: those two languages are not related it would have been better to have two PRs for things like that.

From my point the Lua part looks ok. I'm not sure about LaTeX. Can ` command_name` only occur in command invokation or also in command definition (probably not `command_name` there. @clason

**Review by theHamsta — APPROVED** [review]

File: queries/lua/highlights.scm (hunk 1 above)
Comment (clason) [reply queries/lua/highlights.scm:175]: It's also used in definitions, so this needs to be made more targeted (distinguish based on `field`).

**Review by clason — CHANGES_REQUESTED** [review]
LaTeX change for `command_name` is too broad; this will also apply the capture to definitions.

**Comment by tomtomjhj** [comment]
Thanks for the feedback. Removed the commit for L<sup>A</sup>T<sub>E</sub>X. I'll try other approaches later.

**Review by theHamsta — APPROVED** [review]

**Review by clason — APPROVED** [review]

**Comment by clason** [comment]
Speaking of which: we should probably sync with the new upstream queries: https://github.com/MunifTanjim/tree-sitter-lua/tree/main/queries

(After that, I'll update the bundled core queries.)
